service/src/main/java/tds/assessment/repositories/impl/AssessmentMapper.java

```diff
@@ -75,63 +49,92 @@
-        assessment.setSelectionAlgorithm(Algorithm.fromType((String) assessmentRow.get("selectionalgorithm")));
-        assessment.setStartAbility((float) assessmentRow.get("startAbility"));
-        assessment.setSubject((String) assessmentRow.get("subject"));
+        assessment.setKey(assessmentRow.getString("assessmentSegmentKey"));
+        assessment.setAssessmentId(assessmentRow.getString("assessmentSegmentId"));
+        assessment.setSelectionAlgorithm(assessmentRow.getAlgorithm("selectionalgorithm"));
+        assessment.setStartAbility(assessmentRow.getFloat("startAbility"));
+        assessment.setSubject(assessmentRow.getString("subject"));
+        assessment.setAbilityIntercept(assessmentRow.getFloat("abilityintercept"));
+        assessment.setAbilitySlope(assessmentRow.getFloat("abilityslope"));
+        assessment.setFieldTestStartDate(assessmentRow.getJodaInstantFromTimestamp("ftstartdate"));
+        assessment.setFieldTestEndDate(assessmentRow.getJodaInstantFromTimestamp("ftenddate"));
+        assessment.setAbilitySlope(assessmentRow.getFloat("abilityslope"));
+        assessment.setAbilityIntercept(assessmentRow.getFloat("abilityintercept"));
+        assessment.setAccommodationFamily(assessmentRow.getString("accommodationfamily"));
+        assessment.setMaxOpportunities(assessmentRow.getInt("maxopportunities"));
+
         assessment.setSegments(assessmentSegments);
 
         return Optional.of(assessment);
     }
 
-    private class SegmentInformationHolder {
-        private Segment segment;
-        private List<ItemProperty> languages = new ArrayList<>();
+    private void updateSegmentWithRowData(Segment segment, ResultRow row, boolean isAssessmentRow) {
+        String assessmentKey = isAssessmentRow ? row.getString("assessmentSegmentKey") : row.getString("assessmentKey");
+        int position = isAssessmentRow ? 1 : row.getInt("segmentPosition");
+
+        segment.setSegmentId(row.getString("assessmentSegmentId"));
+        segment.setAssessmentKey(assessmentKey);
+        segment.setSelectionAlgorithm(row.getAlgorithm("selectionalgorithm"));
+        segment.setMinItems(row.getInt("minItems"));
+        segment.setMaxItems(row.getInt("maxItems"));
+        segment.setFieldTestMinItems(row.getInt("fieldTestMinItems"));
+        segment.setFieldTestMaxItems(row.getInt("fieldTestMaxItems"));
+        segment.setStartAbility(row.getFloat("startAbility"));
+        segment.setPosition(position);
+        segment.setSubject(row.getString("subject"));
+
+        //Add the language
+        String propName = row.getString("propname");
+        if (propName != null) {
+            ItemProperty language = new ItemProperty(
+                propName,
+                row.getString("propvalue"),
+                row.getString("propdescription")
+            );
+            segment.getLanguages().add(language);
+        }
+    }
+
+    private class ResultRow {
```

Comment:
I added this class because doing the casting when the value in the map is null will cause a NPE.  Fortunately, the data we have locally and in the deployment db happens to be not null (though the columns allow null).  This seemed like the cleanest way to fetch.